platform/viewer/src/index.js

```diff
@@ -17,11 +17,46 @@ import ReactDOM from 'react-dom';
  */
 import loadDynamicImports from './pluginImports.js';
 
-loadDynamicImports().then(() => {
+const loadDynamicConfigPromise = async () => {
+  let query = new URLSearchParams(window.location.search)
+  let configUrl = query.get('configUrl')
+
+  if (!configUrl) {
+    // Handle OIDC redirects
+    const obj = JSON.parse(
+      sessionStorage.getItem('ohif-redirect-to')
+    );
+    if (obj) {
+      const query = new URLSearchParams(obj.search)
+      configUrl = query.get('configUrl')
+    }
+  }
+
+  if (configUrl) {
+    const response = await fetch(configUrl)
```

Comment:
This leaves some nasty cross-site scripting (CSS) issues.  I'd like to see that the configUrl requires matching a given URL path as defined in the original config file.  Something like a flag in the base configuration whose value is a regular expression defining the allowed paths.  Then, in an example configuration make it a hard URL based path.  You could even add an automatic prefix to it, and a default path, eg  prefixConfigUrl: "./dynamicConfig/" with a prefixCheck of something like /[a-zA-Z0-9\-]*\.json/

Reply:
@Ouwen Does the recent push handle Bill's comment above?

Reply:
@sedghi, the recent push does not account for Bill's comment. There are some other issues with the Access-Control-Headers not being set properly when this dynamic config is used (separate issue).

Regarding cross-site scripting, my understanding is that it should be managed by cross-origin policy rather than forcing the prefix in javascript. This way, if someone is trying to send a POST or PUT request to a different domain via some type of injection it would be stopped by cross-origin.

So `cross-origin = 'same-origin'` is safest.

Happy to still implement the prefix check if needed. In the meantime I'm looking into the access control issue.

Reply:
Honestly we can't rely the user to setup CORS correctly. So I would prefer what Bill suggested above (to have some regular expression test on the config to throw error on non allowed addresses)